src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -405,7 +411,17 @@ private Object readResolve() {
         @Override public String getDisplayName() {
             // TODO more generic to check whether FlowExecution.owner.executable is a ModelObject
             Run<?,?> r = runForDisplay();
-            return r != null ? Messages.ExecutorStepExecution_PlaceholderTask_displayName(r.getFullDisplayName()) : Messages.ExecutorStepExecution_PlaceholderTask_displayName_unknown();
+            if (r != null) {
+                String runDisplayName = r.getFullDisplayName();
+                String enclosingLabel = getEnclosingLabel();
```

Comment:
Unfortunately as per [JENKINS-45553](https://issues.jenkins-ci.org/browse/JENKINS-45553), this is also called by `getFullDisplayName`, which `Queue.maintain` calls frequently, and there is not yet an alternate API to be used for this purpose. Options:

* override `getFullDisplayName` to omit the label, and hope there are no important UI uses (`executorCell` does its own thing, but would need to check the queue widget)
* define the core API and use that here (adding `@Override` only when baseline bumped)
* somehow determine dynamically whether we are holding the `Queue.lock` (guess this would need reflection to obtain that field, followed by `isHeldByCurrentThread`)

CC @svanoort

Reply:
cf. [JENKINS-36547](https://issues.jenkins-ci.org/browse/JENKINS-36547)